pkg/controller/configmap_ca_injector/controller.go

```diff
@@ -0,0 +1,172 @@
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileConfigMapInjector{}
+
+type ReconcileConfigMapInjector struct {
+	client client.Client
+	scheme *runtime.Scheme
+	status *statusmanager.StatusManager
+}
+
+// Reconcile expectes requests to refers to configmaps of two different types.
+// 1. a configmap named trusted-ca-bundle in namespace openshift-config-managed and will ensure that all configmaps with the label
+// config.openshift.io/inject-trusted-cabundle = true have the certificate information stored in trusted-ca-bundle's ca-bundle.crt entry.
+// 2. a configmap in any namespace with the label config.openshift.io/inject-trusted-cabundle = true and will insure that it contains the ca-bundle.crt
+// entry in the configmap named trusted-ca-bundle in namespace openshift-config-managed.
+func (r *ReconcileConfigMapInjector) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("Reconciling configmap from  %s/%s\n", request.Name, request.Namespace)
+
+	trustedCAbundleConfigMap := &corev1.ConfigMap{}
+	trustedCAbundleConfigMapName := types.NamespacedName{
+		Namespace: names.TRUST_BUNDLE_CONFIGMAP_NS,
+		Name:      names.TRUST_BUNDLE_CONFIGMAP,
+	}
+	err := r.client.Get(context.TODO(), trustedCAbundleConfigMapName, trustedCAbundleConfigMap)
+	if err != nil {
+		log.Println(err)
+		return reconcile.Result{}, err
+	}
+	trustedCAbundleData, err := r.validateTrustBundleConfigMap(trustedCAbundleConfigMap)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	//build a list of configMaps
+	configMapsToChange := []corev1.ConfigMap{}
+
+	//The trusted-ca-bundle changed
+	if request.Name == names.TRUST_BUNDLE_CONFIGMAP && request.Namespace == names.TRUST_BUNDLE_CONFIGMAP_NS {
+
+		configMapList := &corev1.ConfigMapList{}
+		selector := labels.Set(map[string]string{names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION: "true"}).AsSelector()
+		err = r.client.List(context.TODO(), &client.ListOptions{LabelSelector: selector}, configMapList)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		configMapsToChange = configMapList.Items
+	} else {
+		//get the requested object
+		requestedCAbundleConfigMap := &corev1.ConfigMap{}
+		requestedCAbundleConfigMapName := types.NamespacedName{
+			Namespace: request.Namespace,
+			Name:      request.Name,
+		}
+		err = r.client.Get(context.TODO(), requestedCAbundleConfigMapName, requestedCAbundleConfigMap)
+		if err != nil {
+			log.Println(err)
+			return reconcile.Result{}, err
+		}
+		configMapsToChange = append(configMapsToChange, *requestedCAbundleConfigMap)
+	}
+
+	for _, configMap := range configMapsToChange {
+		if configMap.Data == nil {
+			configMap.Data = map[string]string{names.TRUST_BUNDLE_CONFIGMAP_KEY: string(trustedCAbundleData)}
+		} else {
+			configMap.Data[names.TRUST_BUNDLE_CONFIGMAP_KEY] = string(trustedCAbundleData)
+		}
+		err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
Suggestion: gather errors in an array, try to update all configmaps, then handle a non-zero `errors` array. That way a single failure won't cause all others to block.

I suggest also adding an error threshold - if `len(errors) > 5`, then just give up.

Reply:
fixed